extensions/tags/src/Filter/PostTagFilter.php

```diff
@@ -11,18 +11,23 @@
 
 use Flarum\Filter\FilterInterface;
 use Flarum\Filter\FilterState;
+use Flarum\Filter\ValidateFilterTrait;
 
 class PostTagFilter implements FilterInterface
 {
+    use ValidateFilterTrait;
+
     public function getFilterKey(): string
     {
         return 'tag';
     }
 
-    public function filter(FilterState $filterState, string $filterValue, bool $negate)
+    public function filter(FilterState $filterState, $filterValue, bool $negate)
     {
+        $ids = $this->asIntArray($filterValue);
+
         $filterState->getQuery()
             ->join('discussion_tag', 'discussion_tag.discussion_id', '=', 'posts.discussion_id')
-            ->where('discussion_tag.tag_id', $negate ? '!=' : '=', $filterValue);
+            ->where('discussion_tag.tag_id', $negate ? '!=' : '=', $ids);
```

Comment:
If this is an intArray, shouldn't it be `whereIn`?

Reply:
nice catch! technically I should have kept it an `int` rather than an `array<int>`. But nothing wrong with allowing filtering per many tags.